### PR TITLE
SAK-50029 WC SakaiGrades obfuscate grades on default page load

### PR DESCRIPTION
**Pull request by ern**
https://sakaiproject.atlassian.net/browse/SAK-50029

### CODE REVIEW

**Comment by ern** [comment]
Sample view when page loads, clicking anywhere on the grades will reveal (remove blur) grades.
image

File: webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js [hunk 1]
```diff
@@ -7,7 +7,10 @@ import { ASSIGNMENT_A_TO_Z, ASSIGNMENT_Z_TO_A, COURSE_A_TO_Z
 
 export class SakaiGrades extends SakaiPageableElement {
 
-  static properties = { _i18n: { state: true } };
+  static properties = {
+    _i18n: { state: true },
+    secret: { type: Boolean },
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js:12]:
```suggestion
    secret: { type: Boolean },
    _i18n: { state: true },
```

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js:12]: I always put the actual attributes first, and the state (private) type stuff afterwards.

File: webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js [hunk 2]
```diff
@@ -76,6 +79,16 @@ export class SakaiGrades extends SakaiPageableElement {
     this.repage();
   }
 
+  firstUpdated() {
+    if (this.secret) {
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js:83]:
```suggestion

    if (this.secret) {
```

File: webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js [hunk 3]
```diff
@@ -206,6 +219,9 @@ export class SakaiGrades extends SakaiPageableElement {
           text-align: right;
           align-items: center;
         }
+        .blurred {
+            filter: blur(3px);
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/packages/sakai-grades/src/SakaiGrades.js:223]:
```suggestion
          filter: blur(3px);
```

File: webcomponents/tool/src/main/frontend/packages/sakai-widgets/src/SakaiGradesWidget.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ export class SakaiGradesWidget extends SakaiDashboardWidget {
 
     return html`
 
-      <sakai-grades
+      <sakai-grades secret
         user-id="${ifDefined(this.userId ? this.userId : undefined)}"
         site-id="${ifDefined(this.siteId ? this.siteId : undefined)}"
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/packages/sakai-widgets/src/SakaiGradesWidget.js:21]:
```suggestion
      <sakai-grades
        user-id="${ifDefined(this.userId)}"
        site-id="${ifDefined(this.siteId)}"
        secret
```

**Review by adrianfish — CHANGES_REQUESTED** [review]

**Comment by ern** [comment]
Updated screenshot
image

**Pull request merged**